streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -640,6 +644,78 @@ public void shouldRespectCommitNeeded() {
         assertFalse(task.commitNeeded());
     }
 
+    @Test
+    public void shouldRestorePartitionTimeAfterRestartWithEosDisabled() {
+        task = createStatelessTask(createConfig(false));
+        task.initializeStateStores();
+        task.initializeTopology();
+
+        task.addRecords(partition1, singletonList(getConsumerRecord(partition1, DEFAULT_TIMESTAMP)));
+
+        task.process();
+
+        task.commit();
+        assertTrue(task.decodeTimestamp(consumer.committed(partition1).metadata())
+                   == DEFAULT_TIMESTAMP);
+        // reset times here to artificially represent a restart
+        task.resetTimes();
+        assertTrue(task.partitionTime(partition1) == RecordQueue.UNKNOWN);
+
+        task.initializeTaskTime();
+        assertTrue(task.partitionTime(partition1) == DEFAULT_TIMESTAMP);
+        assertTrue(task.streamTime() == DEFAULT_TIMESTAMP);
+    }
+
+    @Test
+    public void shouldRestorePartitionTimeAfterRestartWithEosEnabled() {
+        task = createStatelessTask(createConfig(true));
+        task.initializeStateStores();
+        task.initializeTopology();
+
+        task.addRecords(partition1, singletonList(getConsumerRecord(partition1, DEFAULT_TIMESTAMP)));
+
+        task.process();
+        task.commit();
+
+        //since consumer is mock, there is no real broker
+        //so we need to manually commit the information to stimulate broker
+        final Map<TopicPartition, OffsetAndMetadata> offsetMap = new HashMap<>();
+        final String encryptedMetadata = task.encodeTimestamp(DEFAULT_TIMESTAMP);
+        offsetMap.put(partition1, new OffsetAndMetadata(DEFAULT_TIMESTAMP, encryptedMetadata));
+        consumer.commitSync(offsetMap);
+        assertTrue(task.decodeTimestamp(consumer.committed(partition1).metadata())
+                   == DEFAULT_TIMESTAMP);
```

Comment:
This whole block (L680-687) can be remove -- there is no need for the test to commit anything in addition. Note that the `assert` above tests the previous `commitSync()` what is not useful, as test-code should not test other test-code :)